scripts/get_maintainer.py

```diff
@@ -88,9 +89,39 @@ def parse_maintainers():
     return subsystems
 
 
+# If @patchset is a patchset files and contains 2 patches or more, write
+# individual patches to temporary files and return the paths.
+# Otherwise return [].
+def split_patchset(patchset):
+    psname = os.path.basename(patchset).replace('.', '_')
+    patchnum = 0
+    of = None
+    ret = []
+    try:
+        with open(patchset, "r") as f:
+            for line in f:
+                match = re.search(PATCH_START, line)
+                if match:
+                    # New patch found: create new file
+                    patchnum += 1
+                    prefix = "{}_{}_".format(patchnum, psname)
+                    of = tempfile.NamedTemporaryFile(mode="w", prefix=prefix,
+                                                     suffix=".patch",
+                                                     delete=False)
+                    ret += [of.name]
+                of.write(line)
+    except Exception:
```

Comment:
What are the exceptions that you suppress? I guess there are more specific exceptions to catch? ? `OSError`?

Reply:
Well ideally I'd like to simply ignore errors when `patchset` does not exist or is a directory. Because at this point we don't know what is the argument. If it's not a patch/patchset we'll just match it against the patterns in MAINTAINERS.
Now, if an error occurs with the temporary file (`of`), I should normally report it.
Maybe it would be better to do move the input file opening up and leave the rest of the processing without `try/except` like so?
```python
try:
    f = open(...)
except OSError:
    return []
for line in f:
    ...
```

Reply:
That works too, anyhow this is just an utility script, it's not a thing running on a device, so don't overwork it. I usually try to avoid generic exception and add the ones that I've seen when writing / testing my script. But as I said, don't overwork it.